plugin/cache/cache.go

```diff
@@ -162,8 +163,9 @@ func (w *ResponseWriter) WriteMsg(res *dns.Msg) error {
 	if hasKey && duration > 0 {
 		if w.state.Match(res) {
 			w.set(res, key, mt, duration)
-			cacheSize.WithLabelValues(w.server, Success).Set(float64(w.pcache.Len()))
-			cacheSize.WithLabelValues(w.server, Denial).Set(float64(w.ncache.Len()))
+			// TODO better way to do this without joins?
```

Comment:
IIUC "Zones" is a hardcoded list (set here: https://github.com/coredns/coredns/blob/master/plugin/cache/setup.go#L187)

So I don't see any inherently better thing to do than using a separator (e.g. one gauge per zone doesn't sound right).

If you're concerned about performance of joining, perhaps you could compute this label only once on cache instantiation? (this requires doing that in setup.go which is a bit awkward I admit).

Reply:
addressed via https://github.com/DataDog/coredns/pull/69/commits/71f091455e71c105206c2144e41315b47a688ca6